metrics.go

```diff
@@ -148,55 +203,72 @@ var (
 		},
 		[]string{"user", "cluster", "cluster_user", "replica", "cluster_node"},
 	)
 	cacheHitFromConcurrentQueries = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
-			Name: "cache_hit_concurrent_query_total",
-			Help: "The amount of cache hits after having awaited concurrently executed queries",
+			Namespace: namespace,
+			Name:      "cache_hit_concurrent_query_total",
+			Help:      "The amount of cache hits after having awaited concurrently executed queries",
 		},
 		[]string{"cache", "user", "cluster", "cluster_user"},
 	)
 
 	cacheMissFromConcurrentQueries = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
-			Name: "cache_miss_concurrent_query_total",
-			Help: "The amount of cache misses, even if previously reported as queries available in the cache, after having awaited concurrently executed queries",
+			Namespace: namespace,
+			Name:      "cache_miss_concurrent_query_total",
+			Help:      "The amount of cache misses, even if previously reported as queries available in the cache, after having awaited concurrently executed queries",
 		},
 		[]string{"cache", "user", "cluster", "cluster_user"},
 	)
 	killedRequests = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
-			Name: "killed_request_total",
-			Help: "The number of requests killed by proxy",
+			Namespace: namespace,
+			Name:      "killed_request_total",
+			Help:      "The number of requests killed by proxy",
 		},
 		[]string{"user", "cluster", "cluster_user", "replica", "cluster_node"},
 	)
 	timeoutRequest = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
-			Name: "timeout_request_total",
-			Help: "The number of timed out requests",
+			Namespace: namespace,
+			Name:      "timeout_request_total",
+			Help:      "The number of timed out requests",
 		},
 		[]string{"user", "cluster", "cluster_user", "replica", "cluster_node"},
 	)
 	configSuccess = prometheus.NewGauge(prometheus.GaugeOpts{
-		Name: "config_last_reload_successful",
-		Help: "Whether the last configuration reload attempt was successful.",
+		Namespace: namespace,
+		Name:      "config_last_reload_successful",
+		Help:      "Whether the last configuration reload attempt was successful.",
 	})
 	configSuccessTime = prometheus.NewGauge(prometheus.GaugeOpts{
-		Name: "config_last_reload_success_timestamp_seconds",
-		Help: "Timestamp of the last successful configuration reload.",
+		Namespace: namespace,
+		Name:      "config_last_reload_success_timestamp_seconds",
+		Help:      "Timestamp of the last successful configuration reload.",
 	})
 	badRequest = prometheus.NewCounter(prometheus.CounterOpts{
-		Name: "bad_requests_total",
-		Help: "Total number of unsupported requests",
+		Namespace: namespace,
+		Name:      "bad_requests_total",
+		Help:      "Total number of unsupported requests",
 	})
-)
+}
+
+func getNamespace(cfg *config.Config) string {
+	namespace := ""
+	if cfg.EnableMetricNamespace {
+		namespace = "chproxy"
```

Comment:
IMO, this field can be configurable. This `chproxy` value can be the default. If someone needs to use a different value than the default, they will be able to set it in the config.

Reply:
Actually, I agree with you. Would it be better to use another option MetricNamespace `string`?

Reply:
Or providing EnableMetricNamespace `bool` and `MetricNamespace `string` together?

Reply:
Yes. I think so